tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/arrays/ByteArray.java

```diff
@@ -258,4 +258,30 @@ public static ByteArray concat(ByteArray... arrays) {
         }
         return concatArray;
     }
+
+    /**
+     * Extracts a slice of elements from a given {@link ByteArray}, creating a new {@link ByteArray} instance.
+     *
+     *
+     * @param array
+     *     The {@link ByteArray} from which to extract the slice.
+     * @param offset
+     *     The starting index from which to begin the slice, inclusive.
+     * @param length
+     *     The number of elements to include in the slice.
+     * @return A new {@link ByteArray} instance representing the specified slice of the original array.
+     * @throws IllegalArgumentException
+     *     if the specified slice is out of the bounds of the original array.
+     */
+    public static ByteArray slice(ByteArray array, int offset, int length) {
```

Comment:
Does this method need to be `static`. I pictured this utility as an instance method instead. 

```java
var mySlice = myBuffer.slice(offset, size);
```

What do you think?

Reply:
this is now modified to: ```         FloatArray slice = c.slice(256, numElements);  ```